api/server.go

```diff
@@ -42,55 +43,71 @@ func writeResponse(w http.ResponseWriter, status int, response *Response) {
 }
 
 func (a *API) createLoadAgentHandler(w http.ResponseWriter, r *http.Request) {
-	var config loadtest.Config
-	err := json.NewDecoder(r.Body).Decode(&config)
+	var data struct {
+		LoadTestConfig   loadtest.Config
+		ControllerConfig json.RawMessage
```

Comment:
Maybe we can specify both simplecontroller and simulcontroller fields here with omitempty set to true ? So at one time, a user will only set one controller. But still we will be able to apply json validation to it.

Reply:
Good idea 👍